apache/zeppelin
# [ZEPPELIN-1320] Run zeppelin interpreter process as web front end user

### What is this PR for?
While running a Notebook using shell, spark, python uses same user as which zeppelin server is running. Which means these interprets have same permission on file system as zeppelin server.
IMO users should be able to impersonate themselves as a complete security system.

### What type of PR is it?
[Improvement]

### Todos
* [x] - FIX NPEs

### What is the Jira issue?
* [ZEPPELIN-1320](https://issues.apache.org/jira/browse/ZEPPELIN-1320)

### How should this be tested?
 - Enable shiro auth in shiro.ini
 - Add ssh key for the same user you want to try and impersonate (say user1).
```
adduser user1
ssh-keygen
ssh user1@localhost mkdir -p .ssh
cat ~/.ssh/id_rsa.pub | ssh user1@localhost 'cat >> .ssh/authorized_keys'
```
 - Start zeppelin server, try and run following in paragraph in a notebook
 - Go to interpreter setting page, and enable "User Impersonate" in any of the interpreter (in my example its shell interpreter)

```
%sh
whoami
```

Check that it should run as new user, i.e. "user1"

### Screenshots (if appropriate)

![user impersonate](https://cloud.githubusercontent.com/assets/674497/17700306/d9294e80-63e2-11e6-8be8-1c4b84ef5aee.gif)



### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? yes
